File: pos-app/create-test-employees.js
/**
 * Create Test Employees
 * Run this script to populate the database with test employees
 * Usage: node create-test-employees.js
 */

const Database = require('better-sqlite3');
const bcrypt = require('bcryptjs');
const path = require('path');
const os = require('os');

// Get database path (same as Electron app)
const userDataPath = path.join(os.homedir(), 'Library', 'Application Support', 'happy-place-pos');
const dbPath = path.join(userDataPath, 'pos.db');
console.log('Database path:', dbPath);

const db = new Database(dbPath);

// Create test employees
async function createTestEmployees() {
  console.log('Creating test employees...');

  const employees = [
    {
      backend_id: 1,
      email: 'admin@happyplace.co.ke',
      full_name: 'Admin User',
      role: 'admin',
      password: 'admin123'
    },
    {
      backend_id: 2,
      email: 'manager@happyplace.co.ke',
      full_name: 'Manager User',
      role: 'manager',
      password: 'manager123'
    },
    {
      backend_id: 3,
      email: 'cashier1@happyplace.co.ke',
      full_name: 'Cashier One',
      role: 'cashier',
      password: 'cashier123'
    }
  ];

  const stmt = db.prepare(`
    INSERT OR REPLACE INTO employees 
    (backend_id, email, full_name, role, password_hash, permissions, is_active, last_synced_at, created_at, updated_at)
    VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
  `);

  for (const emp of employees) {
    // Hash password
    const passwordHash = await bcrypt.hash(emp.password, 10);
    const now = new Date().toISOString();

    stmt.run(
      emp.backend_id,
      emp.email,
      emp.full_name,
      emp.role,
      passwordHash,
      JSON.stringify({}),
      1, // is_active
      now,
      now,
      now
    );

    console.log(`✅ Created: ${emp.email} (password: ${emp.password})`);
  }

  console.log('\n✅ Test employees created successfully!');
  console.log('\nYou can now login with:');
  console.log('- admin@happyplace.co.ke / admin123');
  console.log('- manager@happyplace.co.ke / manager123');
  console.log('- cashier1@happyplace.co.ke / cashier123');

  db.close();
}

createTestEmployees().catch(console.error);

File: backend/create_test_employees.py
"""
Create test employees directly in PostgreSQL database
"""
import sys
import os

# Add backend to path
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

from app import create_app
from models import db
from models.database_models import Employee
from werkzeug.security import generate_password_hash

app = create_app()

with app.app_context():
    # Check existing employees
    existing = Employee.query.all()
    print(f"\n📋 Current employees: {len(existing)}")
    
    if len(existing) > 0:
        print("\n✅ Employees already exist:")
        for emp in existing:
            print(f"  • {emp.email} - {emp.full_name} ({emp.role})")
        print("\nNo action needed.")
    else:
        print("\n🌱 Creating test employees...")
        
        # Admin
        admin = Employee(
            email='admin@happyplace.com',
            password_hash=generate_password_hash('admin123'),
            full_name='Administrator',
            role='admin',
            active=True
        )
        db.session.add(admin)
        
        # Manager
        manager = Employee(
            email='manager1@happyplace.co.ke',
            password_hash=generate_password_hash('manager123'),
            full_name='Jane Manager',
            role='manager',
            active=True
        )
        db.session.add(manager)
        
        # Cashier
        cashier = Employee(
            email='cashier1@happyplace.co.ke',
            password_hash=generate_password_hash('cashier123'),
            full_name='John Cashier',
            role='cashier',
            active=True
        )
        db.session.add(cashier)
        
        db.session.commit()
        
        print("✅ Created 3 test employees:")
        print("  • admin@happyplace.com / admin123 (admin)")
        print("  • manager1@happyplace.co.ke / manager123 (manager)")
        print("  • cashier1@happyplace.co.ke / cashier123 (cashier)")
    
    print()
